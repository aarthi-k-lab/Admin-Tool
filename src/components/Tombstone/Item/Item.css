.item {
  border-right: dashed 0.0625rem var(--grey-350);
  box-sizing: border-box;
  display: inline-flex;
  flex-direction: column;
  padding: 0 1.25rem;
  padding-bottom: 0.8125rem;
  padding-top: 0.75rem;
  width: -webkit-fill-available;
  vertical-align: center;
}

.title {
  align-self: auto;
  color: var(--grey-590);
  font-size: 0.9em;
  height: 1em;
  display: contents;
}

.content {
  color: var(--grey-850);
  font-size: 0.9em;
  font-weight: bold;
  height: 1em;
  text-transform: uppercase;
  display: block;
}
