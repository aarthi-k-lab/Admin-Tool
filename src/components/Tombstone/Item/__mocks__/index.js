const Item = jest.fn(() => null);
export default Item;
