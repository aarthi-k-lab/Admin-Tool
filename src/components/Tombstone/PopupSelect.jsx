import React from 'react';
import { RFD, COLLATERAL, REASONABLE_EFFECT } from '../../constants/loanInfoComponents';
import RFDContent from './TombstoneComponents/RFDContent';
import CollateralContent from './TombstoneComponents/CollateralContent';
import ReasonableEffort from './TombstoneComponents/ReasonableEffort';

const tombstonePopupMap = {
  [RFD]: <RFDContent />,
  [COLLATERAL]: <CollateralContent />,
  [REASONABLE_EFFECT]: <ReasonableEffort />,
};

function getTombstonePopup(data) {
  return tombstonePopupMap[data];
}

export default getTombstonePopup;
