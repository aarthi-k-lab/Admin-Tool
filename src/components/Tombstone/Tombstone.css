.tombstone {
  align-items: center;
  background-color: var(--grey-100);
  border-bottom: solid 0.0625rem var(--grey-400);
  box-sizing: border-box;
}

.more-icon-button {
  box-sizing: border-box;
  display: inline-flex;
  flex-direction: column;
  justify-content: center;
  height: fit-content;
  vertical-align: inherit;
}

.more-icon {
  color: var(--grey-850);
}

.spacer {
  flex-grow: 1;
}

.icon.icon {
  color: var(--grey-900);
  font-size: 1.125rem;
}

.menuItemRoot.menuItemRoot {
  height: fit-content;
  max-height: 70%;
  width: fit-content;
  overflow-y: scroll;
  margin-top: 5rem;
  right: 0;
}

.menuItemContent {
  font-weight: bold;
  font-size: 1.2rem;
}

.menuItem.menuItem {
  cursor: default;
  opacity: 1;
}

.itemTd {
  border-right: dashed 0.0625rem var(--grey-350);
  text-align: center;
  vertical-align: center;
  flex-direction: column;
  height: 100%;
  overflow: auto;
}

.tombstone-table {
  height: 100%;
}

.highlight {
  background-color: red;
  border-right: dashed 0.0625rem var(--grey-350);
  text-align: center;
  vertical-align: center;
  flex-direction: column;
  height: 100%;
  overflow: auto;
}