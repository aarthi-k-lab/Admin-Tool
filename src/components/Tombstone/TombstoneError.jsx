import React from 'react';
import ErrorIcon from '@material-ui/icons/Error';
import './TombstoneLoader.css';

function TombstoneError() {
  return (
    <section styleName="tombstone-loader">
      <ErrorIcon fontSize="large" />
    </section>
  );
}

export default TombstoneError;
