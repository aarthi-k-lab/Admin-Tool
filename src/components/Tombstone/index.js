export { default } from './Tombstone';
export { default as TombstoneLoader } from './TombstoneLoader';
export { default as TombstoneError } from './TombstoneError';
