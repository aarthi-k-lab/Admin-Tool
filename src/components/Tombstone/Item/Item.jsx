import React from 'react';
import PropTypes from 'prop-types';

import './Item.css';

function Item({ content, title }) {
  let contentStyle = 'content';
  if (title === 'Previous Disposition' && content.trim().toLowerCase() === 'send for qc review') {
    contentStyle = 'qCReviewContent';
  }
  return (
    <div styleName="item">
      <span styleName="title">{title}</span>
      <span styleName={contentStyle}>{content}</span>
    </div>
  );
}

Item.defaultProps = {
  content: 'content',
  title: 'title',
};

Item.propTypes = {
  content: PropTypes.string,
  title: PropTypes.string,
};

export default Item;
