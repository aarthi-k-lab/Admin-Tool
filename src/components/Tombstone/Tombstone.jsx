import React from 'react';
import PropTypes from 'prop-types';
import IconButton from '@material-ui/core/IconButton';
import OpenInNewIcon from '@material-ui/icons/OpenInNew';
import MoreIcon from '@material-ui/icons/MoreHoriz';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import { connect } from 'react-redux';
import Item from './Item';
import Selector from '../../state/ducks/tombstone/selectors';
import styles from './Tombstone.css';

class Tombstone extends React.Component {
  constructor(props) {
    super(props);
    const { items } = props;
    const dummy = items;
    const array = [];
    this.state = {
      anchorEl: null,
      menuItem: dummy.slice(array.length, dummy.length),
      tombStoneArray: array,
    };
    this.handleClose = this.handleClose.bind(this);
    this.handleClick = this.handleClick.bind(this);
    this.resize = this.resize.bind(this);
  }

  componentDidMount() {
    window.addEventListener('resize', this.resize.bind(this));
    this.resize();
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.resize.bind(this));
  }

  resize() {
    const { items } = this.props;
    let { tombStoneArray } = this.state;
    const dummy = items;
    const screenWidth = window.innerWidth;
    if (screenWidth < 300) tombStoneArray = items.slice(0, 2);
    else if (screenWidth < 400) tombStoneArray = items.slice(0, 3);
    else if (screenWidth < 500) tombStoneArray = items.slice(0, 4);
    else if (screenWidth < 600) tombStoneArray = items.slice(0, 4);
    else if (screenWidth < 700) tombStoneArray = items.slice(0, 5);
    else if (screenWidth < 800) tombStoneArray = items.slice(0, 5);
    else if (screenWidth < 900) tombStoneArray = items.slice(0, 6);
    else if (screenWidth < 1000) tombStoneArray = items.slice(0, 7);
    else if (screenWidth < 1100) tombStoneArray = items.slice(0, 8);
    else if (screenWidth > 1100) tombStoneArray = items.slice(0, 9);
    this.setState({ tombStoneArray, menuItem: dummy.slice(tombStoneArray.length, dummy.length) });
  }


  handleClose() {
    this.setState({ anchorEl: null });
  }

  handleClick(event) {
    const { items } = this.props;
    const { tombStoneArray } = this.state;
    this.setState({
      anchorEl: event.currentTarget,
      menuItem: items.slice(tombStoneArray.length, items.length),
    });
  }

  render() {
    const { onOpenWindow } = this.props;
    const { tombStoneArray, anchorEl, menuItem } = this.state;
    const open = Boolean(anchorEl);
    let menuDiv = <div />;
    if (menuItem.length > 0) {
      menuDiv = (
        <div styleName="more-icon-button">
          <IconButton onClick={this.handleClick}>
            <MoreIcon styleName="more-icon" />
          </IconButton>
          <Menu
            anchorEl={anchorEl}
            classes={{
              paper: styles.menuItemRoot,
            }}
            id="long-menu"
            onClose={this.handleClose}
            open={open}
            PaperProps={{
              style: {
                maxHeight: menuItem.length * 500,
                width: 300,
                right: 0,
              },
            }}
          >
            {menuItem.map(option => (
              <MenuItem styleName="menuItem">
                <div>
                  {option.title}
                  <br />
                  <span styleName="menuItemContent">
                    {option.content}
                  </span>
                </div>
              </MenuItem>
            ))}
          </Menu>
        </div>
      );
    }

    const tableWidth = tombStoneArray.length > 2 ? '100%' : '20%';
    return (
      <section id="container" styleName="tombstone">
        <table styleName="tombstone-table" width={tableWidth}>
          <tr>
            {Tombstone.getItems(tombStoneArray)}
            <td>{menuDiv}</td>
            <td>
              <div styleName="spacer" />
              <IconButton onClick={onOpenWindow}>
                <OpenInNewIcon styleName="icon" />
              </IconButton>
            </td>
          </tr>
        </table>
      </section>

    );
  }
}

Tombstone.defaultProps = {
  items: [
    {
      title: 'Loan #',
      content: '67845985',
    },
    {
      title: 'Investors',
      content: 'FHA',
    },
    {
      title: 'Title',
      content: 'Content',
    },
  ],
  onOpenWindow: () => { },
};

Tombstone.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      content: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
    }).isRequired,
  ),
  onOpenWindow: PropTypes.func,
};


Tombstone.getItems = function getItems(items) {
  return items.map(({ content, title }) => (
    <td styleName="itemTd">
      <Item key={title} content={content} title={title} />
    </td>
  ));
};

const mapStateToProps = state => ({
  items: Selector.getTombstoneData(state),
});
const TombstoneContainer = connect(
  mapStateToProps,
  null,
)(Tombstone);

const TestHooks = {
  Tombstone,
};

export default TombstoneContainer;
export { TestHooks };
