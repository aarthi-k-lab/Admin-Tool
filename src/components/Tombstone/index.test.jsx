import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { TombstoneLoader, TombstoneError } from '.';
import Item from './Item';
import { TestHooks } from './Tombstone';

jest.mock('./Item');

describe('<Tombstone />', () => {
  beforeEach(() => {
    Item.mockClear();
  });

  test('contains the items passed as props', () => {
    const items = [
      {
        content: 'C1',
        title: 'T1',
      },
      {
        content: 'C2',
        title: 'T2',
      },
    ];
    const wrapper = shallow(<TestHooks.Tombstone items={items} />);
    wrapper.find('WithStyles(ForwardRef(IconButton))').at(0).simulate('Click');
    expect(wrapper.find(Item.name)).toHaveLength(items.length);
    expect(wrapper.find(Item.name).at(0).props().content).toBe(items[0].content);
    expect(wrapper.find(Item.name).at(0).props().title).toBe(items[0].title);
    expect(wrapper.find(Item.name).at(1).props().content).toBe(items[1].content);
    expect(wrapper.find(Item.name).at(1).props().title).toBe(items[1].title);
  });

  test('calls the \'onOpenWindow\' prop', () => {
    const handleOnOpenWindow = jest.fn();
    const wrapper = shallow(<TestHooks.Tombstone onOpenWindow={handleOnOpenWindow} />);
    expect(wrapper.find('WithStyles(ForwardRef(IconButton))').at(0).prop('onClick')).toBe(handleOnOpenWindow);
  });
});

describe('<TombstoneLoader />', () => {
  test('snapshot test', () => {
    const tree = renderer
      .create(<TombstoneLoader />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<TombstoneError />', () => {
  test('snapshot test', () => {
    const tree = renderer
      .create(<TombstoneError />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
