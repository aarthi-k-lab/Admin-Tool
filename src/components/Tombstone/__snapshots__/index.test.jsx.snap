// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TombstoneError /> snapshot test 1`] = `
<section
  className="components-Tombstone-TombstoneLoader__tombstone-loader"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
    />
  </svg>
</section>
`;

exports[`<TombstoneLoader /> snapshot test 1`] = `
<section
  className="components-Tombstone-TombstoneLoader__tombstone-loader"
>
  <div
    className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": "2rem",
        "width": "2rem",
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</section>
`;
